lib/utils/analytics.ts

```diff
@@ -15,27 +15,29 @@ function initiateAnalytics() {
 /**
  * Captures an analytic event
  *
- * @param {string} title - The title of the event
- * @param {string} property - The property of the event
- * @param {string} value - The value of the event
+ * @param title - The title of the event
+ * @param property - The property of the event
+ * @param value - The value of the event
+ * @param userInfo - The user info for the currently logged in user. Undefined if no user is logged in.
+ *
  */
-function useAnalytics() {
-  const { user } = useSupabaseAuth();
+async function captureAnalytics({ title, property, value, userInfo }: AnalyticEvent) {
+  const analyticsObject: Record<string, string> = {};
+
+  analyticsObject[property] = value;
 
-  return {
-    captureAnalytics({ title, property, value }: AnalyticEvent) {
-      const analyticsObject: Record<string, string> = {};
+  // if a user is not logged in, Posthog will generate an anonymous ID
+  if (userInfo) {
+    let userProperties = {};
 
-      analyticsObject[property] = value;
+    const { company, coupon_code, is_open_sauced_member, is_onboarded, role } = userInfo;
 
-      // if a user is not logged in, Posthog will generate an anonymous ID
-      if (user) {
-        posthog.identify(user.user_metadata.sub);
-      }
+    // A pro user is anyone with a role of 50 or higher
```

Comment:
Note: This works for now, but will need updating when we move over to `tier.run`